File: openapi-test-kotlin-server-quarkus-reactive-client-mp-reactive/testapi/model/Pet.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull
import java.time.OffsetDateTime
import java.util.Optional
import kotlin.collections.List
import kotlin.collections.Map

data class Pet(
  @JsonProperty("tags") val tags: Optional<List<String>>,
  @JsonProperty("id") @NotNull val id: String,
  @JsonProperty("status") @NotNull val status: PetStatus,
  @JsonProperty("createdAt") @NotNull val createdAt: OffsetDateTime,
  @JsonProperty("metadata") val metadata: Optional<Map<String, String>>,
  /** Pet name */
  @JsonProperty("name") @NotNull val name: String,
  @JsonProperty("updatedAt") val updatedAt: Optional<OffsetDateTime>
)

File: testers/openapi/scala/spring/testapi/model/Pet.scala
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime
import java.util.Optional

case class Pet(
    @JsonProperty("tags") tags: Optional[java.util.List[String]],
    @JsonProperty("id") id: PetId,
    @JsonProperty("status") status: PetStatus,
    @JsonProperty("createdAt") createdAt: OffsetDateTime,
    @JsonProperty("metadata") metadata: Optional[java.util.Map[String, String]],
    /** Pet name */
    @JsonProperty("name") name: String,
    @JsonProperty("updatedAt") updatedAt: Optional[OffsetDateTime]
)
